### PR TITLE
20 fungible token vote

### PR DESCRIPTION
**Pull request by 0xMazout**
Adding : 

Cast Vote : To spend voting power on a proposal

CRUD Proposal : To manage proposal

Delegate : To Delegate our voting power

Storing datas : 
VotingData : Related to Voting Data proposal , 
Data : Related to Data proposal,
VotingData*ProposalID*AddrSpender* : Related to voting Data user on proposal
proposal : Related to all proposals created by the addr

### CODE REVIEW

**Comment by 0xMazout** [comment]
There is package.json push to correct the bug we find on asprettier windows stuff

**Comment by 0xMazout** [comment]
Currently creating some units tests but we can review the architecture  and the choice made

File: assembly/token/impl.ts [hunk 1]
```diff
@@ -379,3 +381,603 @@ export function transferFrom(stringifyArgs: string): string {
...
+  const amount = args.nextU64();
+
+  if (!ownerAddress.isValid() || !recipientAddress.isValid() || isNaN(amount)) {
+    return '0';
+  }
+
+  const ownerBalance = _balance(ownerAddress);
+  const recipientBalance = _balance(recipientAddress);
+  
+  const delegatedPowerRecipient = new Args(Storage.getOf(recipientAddress, 'DelegatedPower')).nextU64();
```
Comment (SebastienValla) [inline assembly/token/impl.ts:442]: why use args in this case and not  u64(string) ?

File: assembly/token/impl.ts [hunk 2]
```diff
@@ -379,3 +381,603 @@ export function transferFrom(stringifyArgs: string): string {
...
+    return 'Voting : Invalid reason';
+  }
+
+  // Check if the proposal State is Active
+  if (Storage.hasOf(proposalOwner, 'Data'.concat(proposalId))) {
+    let dataProposal = Storage.getOf(proposalOwner, 'Data'.concat(proposalId));
+    let argsproposal = new Args(dataProposal);
+    const owner = argsproposal.nextAddress();
+    const id = argsproposal.nextString();
+    const title = argsproposal.nextString();
```
Comment (SebastienValla) [inline assembly/token/impl.ts:500]: title is never used

**Review by SebastienValla — CHANGES_REQUESTED** [review]

**Comment by Thykof** [comment]
fix #29

**Comment by Thykof** [comment]
this will be done in https://github.com/massalabs/massa-sc-contracts/pull/12

**Pull request closed without merging**